app/controllers/concerns/throttling_manager.rb

```diff
@@ -0,0 +1,33 @@
+module ThrottlingManager
+  extend ActiveSupport::Concern
+
+  included do
+    def decrement_throttle_counts
+      if request.env["rack.attack.throttle_data"]
+        request.env["rack.attack.throttle_data"].each do |name, values|
+          decrement_throttle_count(name, values[:discriminator])
+        end
+      end
+    end
+  end
+
+private
+
+  def decrement_throttle_count(throttle_name, descriminator)
+    last_epoch_time = Time.zone.now.to_i
+    period = Rack::Attack.configuration.throttles[throttle_name].period
+    expires_in = (period - (last_epoch_time % period) + 1).to_i
+    key_name = [
+      "rack::attack",
+      (last_epoch_time / period).to_i,
+      throttle_name,
+      descriminator,
+    ].join(":")
+
+    Rack::Attack.cache.store.decrement(
+      key_name,
+      1,
+      expires_in: expires_in,
```

Comment:
Do we need to specify an expiry for a decrement?

Reply:
Doesn't appear like we do. Tests pass and rate limiting works as expected without